Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -342,44 +342,66 @@ namespace AZ::DocumentPropertyEditor
                     false);
                 return;
             }
-
-            auto containerAttribute = attributes.Find(AZ::Reflection::DescriptorAttributes::Container);
-            if (!containerAttribute.IsNull())
+            else
             {
-                auto container = AZ::Dom::Utils::ValueToTypeUnsafe<AZ::SerializeContext::IDataContainer*>(containerAttribute);
-                m_containers.SetValue(m_builder.GetCurrentPath(), BoundContainer{ container, access.Get() });
-                size_t containerSize = container->Size(access.Get());
-                if (containerSize == 1)
-                {
-                    m_builder.Label("1 element");
-                }
-                else
+                AZStd::string_view labelAttribute = "MISSING_LABEL";
```

Comment:
Is this a bad idea? In other words, would we ever have a legit case where there is no label set below? Otherwise this might draw attention to the issue, potentially helping to debug it.

Reply:
A string view is just a reference to an existing string. So what you're saying here is that `labelAttribute` is a view into some memory that stored the `"MISSING_LABEL"` string somewhere. This operation though...
```
labelAttribute = label.GetString();
```
is not changing the memory where `"MISSING_LABEL"` is stored, but just moving the view to reference whatever position in memory is returned by `label.GetString();`.

This shouldn't be a huge issue since it's all on the stack and does not persist, but for this kind of operation I'd recommend just using a regular `AZStd::string` since it makes all operations clearer and closer to what you'd expect by reading them.

Reply:
I think it's almost certainly programmer error to create a label without an attribute, so this is fine.

And for reference it's fully safe to store a persistent pointer to a string literal (they're stored in a baked literal table, which generally enables `constexpr string_view`)

Reply:
> And for reference it's fully safe to store a persistent pointer to a string literal (they're stored in a baked literal table, which generally enables constexpr string_view)

Oh I didn't know that, good to know :D

Reply:
ok, so does the rest of this all look good to you @NicholasVanSickle , our prodigal architect? :D

Reply:
ok, so does the rest of this all look good to you @NicholasVanSickle , our prodigal architect? :D

Reply:
Looks good to me besides one possible label issue!